UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/InGameDeviceLinker/InGameDeviceLinker.cs

```diff
@@ -0,0 +1,354 @@
+		if (cursorObject.SelectedMaster != null)
+		{
+			if (Select)
+			{
+				SetupGizmosFor(cursorObject.SelectedMaster);
+				CursorLine = GameGizmomanager.AddNewLineStatic(cursorObject.SelectedMaster.gameObject, Vector3.zero,
+					cursorObject.gameObject,
+					Vector3.zero, new Color(1f, 0f, 1f, 1f));
+			}
+
+			else
+			{
+				DeviceLinkMessage.Send(HitGameObject, cursorObject.SelectedMaster.ConType, null);
+			}
+		}
+	}
+
+	/// <summary>
+	/// Tries to spawn at the specified position. Lets you spawn anywhere, even impassable places. Go hog wild!
+	/// </summary>
+	private void TryConnectSelect()
+	{
+		Vector3Int position = cursorObject.transform.position.RoundToInt();
+		var hit = MouseUtils.GetOrderedObjectsUnderMouse()?.FirstOrDefault();
+		if (hit == null) return;
+
+		if (cursorObject.SelectedMaster != null)
+		{
+			if (hit.gameObject == cursorObject.SelectedMaster.gameObject)
+			{
+				// bool Found = false;
+				bool NewMaster = false;
+				// var Masters = HitGameObject.GetComponents<IMultitoolMasterable>();
+				// foreach (var Master in Masters)
+				// {
+				// 	if (Master == cursorObject.SelectedMaster)
+				// 	{
+				// 		Found = true;
+				// 		continue;
+				// 	}
+				//
+				// 	if (Found)
+				// 	{
+				// 		NewMaster = true;
+				// 		CleanupGizmos();
+				// 		cursorObject.SelectedMaster = HitGameObject.GetComponent<IMultitoolMasterable>();
+				// 		if (cursorObject.SelectedMaster != null)
+				// 		{
+				// 			SetupGizmosFor(cursorObject.SelectedMaster);
+				// 			CursorLine = GameGizmomanager.AddNewLineStatic(cursorObject.SelectedMaster.gameObject, Vector3.zero, cursorObject.gameObject,
+				// 				Vector3.zero,new Color(1f,0f,1f,1f));
+				// 		}
+				// 	}
+				// }
+
+				if (NewMaster == false)
+				{
+					cursorObject.SelectedMaster = null;
+					CleanupGizmos();
+				}
+			}
+			else
+			{
+				var Slaves = hit.GetComponents<IMultitoolSlaveable>();
+				foreach (var Slave in Slaves)
+				{
+					if (Slave != null)
+					{
+						if (Slave.ConType != cursorObject.SelectedMaster.ConType) continue;
+						if (Slave.Master == cursorObject.SelectedMaster) //un select
+						{
+							if (CustomNetworkManager.IsServer)
+							{
+								MastersData[((IMultitoolMasterable) cursorObject.SelectedMaster).ConType][
+									(IMultitoolMasterable) cursorObject.SelectedMaster].Remove(Slave);
+							}
+
+							LinkGizmo[Slave].Remove();
+							LinkGizmo.Remove(Slave);
+							Slave.SetMasterEditor(null);
+							DeviceLinkMessage.Send(null, cursorObject.SelectedMaster.ConType,
+								(Slave as Component)?.gameObject);
+						}
+						else if (Slave.Master != cursorObject.SelectedMaster || Slave.Master == null)
+						{
+							Slave.SetMasterEditor(cursorObject.SelectedMaster);
+							LinkGizmo[Slave] = GameGizmomanager.AddNewLineStatic(Slave.gameObject, Vector3.zero,
+								cursorObject.SelectedMaster.gameObject,
+								Vector3.zero, Color.green);
+
+							if (CustomNetworkManager.IsServer)
```

Comment:
for the love of god stop nesting stuff like this
split these into functions and keep if statements reversed.